src/preferences/dialog/dlgprefsound.cpp

```diff
@@ -87,11 +87,10 @@ DlgPrefSound::DlgPrefSound(QWidget* pParent,
             &DlgPrefSound::engineClockChanged);
 
     keylockComboBox->clear();
-    for (int i = 0; i < EngineBuffer::KEYLOCK_ENGINE_COUNT; ++i) {
-        const auto engine = static_cast<EngineBuffer::KeylockEngine>(i);
+    for (const auto engine : EngineBuffer::kKeylockEngines) {
```

Comment:
We actually need an int below in this loop, so there is not much a point of using a range based loop.

Reply:
As pointed out above, I didn't want to do that because I wanted to avoid the cast from int to Keylock engine when I couldn't be sure that the values actually came from Keylockengine in the first place. So this solution is somewhat more typesafe.

Reply:
Actually, the problem is that we're converting to an in so we can use it as a QVariant, but throwing the enum value into QVariant right of the bat would make much more sense. Let me fix that.

Reply:
fixed